include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -2980,6 +3017,84 @@ __pattern_merge(_Tag, _ExecutionPolicy&&, _ForwardIterator1 __first1, _ForwardIt
                                      typename _Tag::__is_vector{});
 }
 
+template <typename _Tag, typename _ExecutionPolicy, typename _It1, typename _Index1, typename _It2, typename _Index2,
+          typename _OutIt, typename _Index3, typename _Comp>
+std::pair<_It1, _It2>
+___merge_path_out_lim(_Tag, _ExecutionPolicy&& __exec, _It1 __it_1, _Index1 __n_1, _It2 __it_2, _Index2 __n_2,
+                      _OutIt __it_out, _Index3 __n_out, _Comp __comp)
+{
+    return __serial_merge_out_lim(__it_1, __it_1 + __n_1, __it_2, __it_2 + __n_2, __it_out, __it_out + __n_out, __comp);
+}
```

Comment:
This is funny :) @MikeDvorskiy you explained to me that the use of indexes in this function is needed to avoid computing the indexes twice. But in the serial implementation you do not use indexes but switch back to the iterators :) which only confirms my impression that `___merge_path_out_lim` should use iterators as its parameters.

Reply:
I told that I was going to avoid twice computation for new output size 
`const _Index3 __n_out = std::min<_Index3>(__n_1 + __n_2, std::ranges::size(__out_r));`

And in parallel version `___merge_path_out_lim`  `__n_1, __n_2, n_out` are used.
As far as we already computed these values in `__pattern_merge`, I just pass ones into `___merge_path_out_lim`.

Reply:
The thing is, you do not need `__n_out` for the serial version. In the serial loop, it's not a problem to check on each iteration if the end of the output is reached.

Reply:
..
Basically, only parallel version of ___merge_path_out_lim needs `const _Index3 __n_out = std::min<_Index3>(__n_1 + __n_2, std::ranges::size(__out_r));` logic.
So, we can move the mentioned logic inside the function. (and pass iterators from the `__pattern_merge`).